packages/plugin-ext/src/main/browser/label-service-main.ts

```diff
@@ -0,0 +1,49 @@
+/********************************************************************************
+ * Copyright (C) 2020 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { LabelServiceMain } from '../../common/plugin-api-rpc';
+import { interfaces } from 'inversify';
+import { Disposable, DisposableCollection } from '@theia/core/lib/common/disposable';
+import { DefaultUriLabelProviderContribution, LabelProviderContribution, ResourceLabelFormatter } from '@theia/core/lib/browser';
+import { ContributionProvider } from '@theia/core/lib/common';
+
+export class LabelServiceMainImpl implements LabelServiceMain {
+    private readonly resourceLabelFormatters = new Map<number, Disposable>();
+    private readonly contributionProvider: ContributionProvider<LabelProviderContribution>;
+
+    constructor(container: interfaces.Container) {
+        this.contributionProvider = container.getNamed(ContributionProvider, LabelProviderContribution);
+    }
+
+    $registerResourceLabelFormatter(handle: number, formatter: ResourceLabelFormatter): void {
+        formatter.priority = true;
```

Comment:
Could you elaborate on this? Is it the same in VS Code?

Reply:
It means that the formatter registered via plugin API will have higher priority than the formotter registered in an extention: https://github.com/eclipse-theia/theia/blob/6eb873187b733baf5cee19f3bba044259d57e778/packages/core/src/browser/label-provider.ts#L293-L297
It comes from vscode:
https://github.com/microsoft/vscode/blob/eaa2d3bb23ce6dc79eadb497ea697e4c192ece6a/src/vs/workbench/api/browser/mainThreadLabelService.ts#L23

Reply:
thank you, copying the comment would help too :D